# Introduction_to_MongoDB
**17 Nov 2022**    
[Lecture link](https://www.youtube.com/watch?v=qiVH8WnTdCU)  


## sql limitation 
1. Need Create  schema.
1. Only structure data can be insert.
1. Search operation problem.

Can be deal by mongodb and then all operation will solve.   
[go to setup](https://cloud.mongodb.com/v2/636e515ef80b866a36df8fda#clusters/edit?filter=starter&fromPathSelector=true) 

**URL COPY FROM MONGO WEBSITE**

In [1]:
# URL COPY FROM MONGO WEBSITE
client = pymongo.MongoClient("mongodb+srv://t20071:<password>@clustor0.rqhqqwa.mongodb.net/?retryWrites=true&w=majority", server_api=ServerApi('1'))
db = client.test

NameError: name 'pymongo' is not defined

Need driver like in sql case 

In [2]:
! pip install pymongo[srv]

In [3]:
import pymongo
client = pymongo.MongoClient("mongodb+srv://t20071:<password>@clustor0.rqhqqwa.mongodb.net/?retryWrites=true&w=majority")
db = client.test
print(db)

Database(MongoClient(host=['ac-paodbyl-shard-00-02.rqhqqwa.mongodb.net:27017', 'ac-paodbyl-shard-00-00.rqhqqwa.mongodb.net:27017', 'ac-paodbyl-shard-00-01.rqhqqwa.mongodb.net:27017'], document_class=dict, tz_aware=False, connect=True, retrywrites=True, w='majority', authsource='admin', replicaset='atlas-g3ag2g-shard-0', tls=True), 'test')


Need to enter the password 

In [4]:
import pymongo
client = pymongo.MongoClient("mongodb+srv://t20071:t20071@clustor0.rqhqqwa.mongodb.net/?retryWrites=true&w=majority")
db = client.test
print(db)

Database(MongoClient(host=['ac-paodbyl-shard-00-02.rqhqqwa.mongodb.net:27017', 'ac-paodbyl-shard-00-00.rqhqqwa.mongodb.net:27017', 'ac-paodbyl-shard-00-01.rqhqqwa.mongodb.net:27017'], document_class=dict, tz_aware=False, connect=True, retrywrites=True, w='majority', authsource='admin', replicaset='atlas-g3ag2g-shard-0', tls=True), 'test')


**27017 = port number**  
NO DNS ISSUES FOUND          
IF FOUND, DO          
```! pip install dnspython```   
before installing pymongo

In [5]:
db1 = client.test1

In [6]:
db2 = client['SHARAD']

NOt reflect on mangobd account

In [7]:
client.list_database_names()

['SHARAD', 'admin', 'local']

in sql = called table             
in mongodb called collection, int que

mysql , mongo        
DB   , DB      
TABLE , COLLECTION      
ROW/RECORD , DOCUMENT     

In [8]:
coll1 = db2['ineuroncollection']

Nothing reflect on cloud mongo

In [9]:
dict1 = {
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited'
}

In [10]:
coll1.insert_one(dict1)

Now,data reflect on mongodb cloud.         
Single or multiple record in mongodb called one document.

In [11]:
dict2 = {
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited',
    'phone_number':'8989969327'
}

In [12]:
coll1.insert_one(dict2)

NOt create schema- good thing in mongoDB     
not define data type,   
Unstructured data intered 

In [13]:
dict3 = {
    'id':'t20071',
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited',
    'phone_number':'8989969327'
}

In [14]:
coll1.insert_one(dict3)

can be updated in mongoDB cloud eady UI like excel

Dict convert into table by                -
Dict keys will be column and values will be row 

In [15]:
list1 = [
        {
            "item": "canvas",
            "qty": 100,
            "size": {"h": 28, "w": 35.5, "uom": "cm"},
            "status": "A",
        },
        {
            "item": "journal",
            "qty": 25,
            "size": {"h": 14, "w": 21, "uom": "cm"},
            "status": "A",
        },
        {
            "item": "mat",
            "qty": 85,
            "size": {"h": 27.9, "w": 35.5, "uom": "cm"},
            "status": "A",
        },
        {
            "item": "mousepad",
            "qty": 25,
            "size": {"h": 19, "w": 22.85, "uom": "cm"},
            "status": "P",
        },
        {
            "item": "notebook",
            "qty": 50,
            "size": {"h": 8.5, "w": 11, "uom": "in"},
            "status": "P",
        },
        {
            "item": "paper",
            "qty": 100,
            "size": {"h": 8.5, "w": 11, "uom": "in"},
            "status": "D",
        },
        {
            "item": "planner",
            "qty": 75,
            "size": {"h": 22.85, "w": 30, "uom": "cm"},
            "status": "D",
        },
        {
            "item": "postcard",
            "qty": 45,
            "size": {"h": 10, "w": 15.25, "uom": "cm"},
            "status": "A",
        },
        {
            "item": "sketchbook",
            "qty": 80,
            "size": {"h": 14, "w": 21, "uom": "cm"},
            "status": "A",
        },
        {
            "item": "sketch pad",
            "qty": 95,
            "size": {"h": 22.85, "w": 30.5, "uom": "cm"},
            "status": "A",
        },
    ]

In [16]:
coll1.insert_many(list1)

succesfully dump in mogno databases

In [17]:
list2 = [{
    'id':'t20071',
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited',
    'phone_number':'8989969327'
},{
    'id':'t20071',
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited',
    'phone_number':'8989969327'
},{
    'id':'t20071',
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited',
    'phone_number':'8989969327'
},{
    'id':'t20071',
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited',
    'phone_number':'8989969327'
},{
    'id':'t20071',
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited',
    'phone_number':'8989969327'
},{
    'id':'t20071',
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited',
    'phone_number':'8989969327'
},{
    'id':'t20071',
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited',
    'phone_number':'8989969327'
},{
    'id':'t20071',
    'Name':'Sharad',
    'email_id':'sharadpardhe08@gmail.com',
    'product':['one neuron', 'tech','kid','drone'],
    'company': 'BBB Private limited',
    'phone_number':'8989969327'
}]

In [18]:
coll1.insert_many(list2)

Easy and effective as compare to sql.

every records need to be insert in dict form

In [19]:
coll1.find()

In [20]:
for i in coll1.find():
    print(i)

{'_id': ObjectId('63762e071521e52836731fd6'), 'item': 'canvas', 'qty': 400, 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd7'), 'item': 'journal', 'qty': 400, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd8'), 'item': 'mat', 'qty': 400, 'size': {'h': 27.9, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'}
{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'}
{'_id': ObjectId('63762e071521e52836731fdb'), 'item': 'paper', 'qty': 400, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'D'}
{'_id': ObjectId('63762e071521e52836731fdc'), 'item': 'planner', 'qty': 400, 'size': {'h': 22.85, 'w': 30, 'uom': 'cm'}, 'status': 'D'}
{'_id': ObjectId('63762e071521e52836731fdd'), 'item': 'postcar

In [21]:
coll1.find({'Name':'Sharad'})

In [22]:
for i in coll1.find({'Name':'Sharad'}):
    print(i)

{'_id': ObjectId('63765d1b7cfa301fe9a8047c'), 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited'}
{'_id': ObjectId('63765d1b7cfa301fe9a8047d'), 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited', 'phone_number': '8989969327'}
{'_id': ObjectId('63765d1b7cfa301fe9a8047e'), 'id': 't20071', 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited', 'phone_number': '8989969327'}
{'_id': ObjectId('63765d1c7cfa301fe9a80489'), 'id': 't20071', 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited', 'phone_number': '8989969327'}
{'_id': ObjectId('63765d1c7cfa301fe9a8048a'), 'id': 't20071', 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'p

In [23]:
for i in coll1.find({'Name':'*har*'}):
    print(i)

NO record availabe 

In [24]:
coll1.find({"Name":{'$in': ['sharad','Sharad','ShaRad']}})
# find id in name  key weather ['sharad','Sharad','ShaRad'] contain or not either of this

In [25]:
for i in coll1.find({"Name":{'$in': ['sharad','Sharad','ShaRad']}}):
    print(i)

{'_id': ObjectId('63765d1b7cfa301fe9a8047c'), 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited'}
{'_id': ObjectId('63765d1b7cfa301fe9a8047d'), 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited', 'phone_number': '8989969327'}
{'_id': ObjectId('63765d1b7cfa301fe9a8047e'), 'id': 't20071', 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited', 'phone_number': '8989969327'}
{'_id': ObjectId('63765d1c7cfa301fe9a80489'), 'id': 't20071', 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited', 'phone_number': '8989969327'}
{'_id': ObjectId('63765d1c7cfa301fe9a8048a'), 'id': 't20071', 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'p

In [26]:
for i in coll1.find({"Name":{'$in': ['sharad','Sharad','ShaRad']},'email_id':'sharadpardhe08@gmail.com'}):
    print(i)

{'_id': ObjectId('63765d1b7cfa301fe9a8047c'), 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited'}
{'_id': ObjectId('63765d1b7cfa301fe9a8047d'), 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited', 'phone_number': '8989969327'}
{'_id': ObjectId('63765d1b7cfa301fe9a8047e'), 'id': 't20071', 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited', 'phone_number': '8989969327'}
{'_id': ObjectId('63765d1c7cfa301fe9a80489'), 'id': 't20071', 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'product': ['one neuron', 'tech', 'kid', 'drone'], 'company': 'BBB Private limited', 'phone_number': '8989969327'}
{'_id': ObjectId('63765d1c7cfa301fe9a8048a'), 'id': 't20071', 'Name': 'Sharad', 'email_id': 'sharadpardhe08@gmail.com', 'p

In [27]:
coll1.find_one()

{'_id': ObjectId('63762e071521e52836731fd6'),
 'item': 'canvas',
 'qty': 400,
 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'},
 'status': 'A'}

In [28]:
coll1.find({"qty" : {'$gy':25}})

In [29]:
for i in coll1.find({"qty" : {'$gt':25}}):
    print(i)

{'_id': ObjectId('63762e071521e52836731fd6'), 'item': 'canvas', 'qty': 400, 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd7'), 'item': 'journal', 'qty': 400, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd8'), 'item': 'mat', 'qty': 400, 'size': {'h': 27.9, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'}
{'_id': ObjectId('63762e071521e52836731fdb'), 'item': 'paper', 'qty': 400, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'D'}
{'_id': ObjectId('63762e071521e52836731fdc'), 'item': 'planner', 'qty': 400, 'size': {'h': 22.85, 'w': 30, 'uom': 'cm'}, 'status': 'D'}
{'_id': ObjectId('63762e071521e52836731fdd'), 'item': 'postcard', 'qty': 45, 'size': {'h': 10, 'w': 15.25, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fde'), 'item': 'sketchb

In [30]:
for i in coll1.find({"qty" : {'$lt':25}}):
    print(i)

In [31]:
for i in coll1.find({"qty" : {'$lte':25}}):
    print(i)

{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'}
{'_id': ObjectId('637652bc1521e52836731fee'), 'item': 'journal', 'qty': 25, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('637652bc1521e52836731ff0'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'}
{'_id': ObjectId('637652f01b9c78260600fd3b'), 'item': 'journal', 'qty': 25, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('637652f01b9c78260600fd3d'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'}
{'_id': ObjectId('637655311b9c78260600fd52'), 'item': 'journal', 'qty': 25, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('637655311b9c78260600fd54'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'}
{'_id': ObjectId('63765d1b7cfa301fe9a80480'), 'item': 'journ

In [32]:
for i in coll1.find({"qty" : {'$gte':25}}): # greater than equal to 
    print(i)

{'_id': ObjectId('63762e071521e52836731fd6'), 'item': 'canvas', 'qty': 400, 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd7'), 'item': 'journal', 'qty': 400, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd8'), 'item': 'mat', 'qty': 400, 'size': {'h': 27.9, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'}
{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'}
{'_id': ObjectId('63762e071521e52836731fdb'), 'item': 'paper', 'qty': 400, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'D'}
{'_id': ObjectId('63762e071521e52836731fdc'), 'item': 'planner', 'qty': 400, 'size': {'h': 22.85, 'w': 30, 'uom': 'cm'}, 'status': 'D'}
{'_id': ObjectId('63762e071521e52836731fdd'), 'item': 'postcar

## update operation 

In [33]:
for i in coll1.find():
    print(i)

{'_id': ObjectId('63762e071521e52836731fd6'), 'item': 'canvas', 'qty': 400, 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd7'), 'item': 'journal', 'qty': 400, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd8'), 'item': 'mat', 'qty': 400, 'size': {'h': 27.9, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'}
{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'}
{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'}
{'_id': ObjectId('63762e071521e52836731fdb'), 'item': 'paper', 'qty': 400, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'D'}
{'_id': ObjectId('63762e071521e52836731fdc'), 'item': 'planner', 'qty': 400, 'size': {'h': 22.85, 'w': 30, 'uom': 'cm'}, 'status': 'D'}
{'_id': ObjectId('63762e071521e52836731fdd'), 'item': 'postcar

Next- want to change name

In [34]:
# Next- want to change name
coll1.update_many({"Name":"Sharad"},{"$set":{"Name":"Sharad Pardhe"}})

In [35]:
for i in coll1.find():
    print(i,'\n')

{'_id': ObjectId('63762e071521e52836731fd6'), 'item': 'canvas', 'qty': 400, 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd7'), 'item': 'journal', 'qty': 400, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd8'), 'item': 'mat', 'qty': 400, 'size': {'h': 27.9, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fdb'), 'item': 'paper', 'qty': 400, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdc'), 'item': 'planner', 'qty': 400, 'size': {'h': 22.85, 'w': 30, 'uom': 'cm'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdd'), 'i

In [36]:
for i in coll1.find().limit(10):
    print(i,'\n')

{'_id': ObjectId('63762e071521e52836731fd6'), 'item': 'canvas', 'qty': 400, 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd7'), 'item': 'journal', 'qty': 400, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd8'), 'item': 'mat', 'qty': 400, 'size': {'h': 27.9, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fdb'), 'item': 'paper', 'qty': 400, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdc'), 'item': 'planner', 'qty': 400, 'size': {'h': 22.85, 'w': 30, 'uom': 'cm'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdd'), 'i

In [37]:
for i in coll1.find({'qty':{"$not":{'$gt':100}}}).limit(10):
    print(i,"\n")

{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fdd'), 'item': 'postcard', 'qty': 45, 'size': {'h': 10, 'w': 15.25, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('637652bc1521e52836731fee'), 'item': 'journal', 'qty': 25, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('637652bc1521e52836731ff0'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('637652bc1521e52836731ff1'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'} 

{'_id': ObjectId('637652bc1521e52836731ff4'), 'item': 'postcard', 'qty': 45, 'size': {'h': 10, 'w': 15.25, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('637652f01b9c78260600fd3b

In [38]:
for i in coll1.find({'qty':{"$not":{'$gt':25}}}).limit(10):
    print(i,"\n")

{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('637652bc1521e52836731fee'), 'item': 'journal', 'qty': 25, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('637652bc1521e52836731ff0'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('637652f01b9c78260600fd3b'), 'item': 'journal', 'qty': 25, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('637652f01b9c78260600fd3d'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('637655311b9c78260600fd52'), 'item': 'journal', 'qty': 25, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('637655311b9c78260600fd54'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('63765d1b7cfa301fe9a8047c'), 

In [39]:
for i in coll1.find({'qty':{"$not":{'$lte':25}}}).limit(10):
    print(i,"\n")

{'_id': ObjectId('63762e071521e52836731fd6'), 'item': 'canvas', 'qty': 400, 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd7'), 'item': 'journal', 'qty': 400, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd8'), 'item': 'mat', 'qty': 400, 'size': {'h': 27.9, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fdb'), 'item': 'paper', 'qty': 400, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdc'), 'item': 'planner', 'qty': 400, 'size': {'h': 22.85, 'w': 30, 'uom': 'cm'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdd'), 'item': 'postcard', 'qty': 45, 'size': {'h': 10, 'w': 15.25, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fde'), 'i

In [40]:
# update qty = 250 where item == journal
coll1.find_one_and_update({'item':'journal'},{'$set': {'qty':250}})

{'_id': ObjectId('63762e071521e52836731fd7'),
 'item': 'journal',
 'qty': 400,
 'size': {'h': 14, 'w': 21, 'uom': 'cm'},
 'status': 'A'}

In [41]:
for i in coll1.find():
    print(i,'\n')

{'_id': ObjectId('63762e071521e52836731fd6'), 'item': 'canvas', 'qty': 400, 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd7'), 'item': 'journal', 'qty': 250, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd8'), 'item': 'mat', 'qty': 400, 'size': {'h': 27.9, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fdb'), 'item': 'paper', 'qty': 400, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdc'), 'item': 'planner', 'qty': 400, 'size': {'h': 22.85, 'w': 30, 'uom': 'cm'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdd'), 'i

In [42]:
coll1.update_many({'qty':{"$gt":50}},{'$set': {'qty': 400}})

In [43]:
for i in coll1.find().limit(10):
    print(i,'\n')

{'_id': ObjectId('63762e071521e52836731fd6'), 'item': 'canvas', 'qty': 400, 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd7'), 'item': 'journal', 'qty': 400, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd8'), 'item': 'mat', 'qty': 400, 'size': {'h': 27.9, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fdb'), 'item': 'paper', 'qty': 400, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdc'), 'item': 'planner', 'qty': 400, 'size': {'h': 22.85, 'w': 30, 'uom': 'cm'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdd'), 'i

### delete operation 

In [44]:
# delete all the record where Name = Sharad Pardhe
coll1.delete_many({"Name":"Sharad Pardhe"})

In [45]:
for i in coll1.find().limit(10):
    print(i,'\n')

{'_id': ObjectId('63762e071521e52836731fd6'), 'item': 'canvas', 'qty': 400, 'size': {'h': 28, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd7'), 'item': 'journal', 'qty': 400, 'size': {'h': 14, 'w': 21, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd8'), 'item': 'mat', 'qty': 400, 'size': {'h': 27.9, 'w': 35.5, 'uom': 'cm'}, 'status': 'A'} 

{'_id': ObjectId('63762e071521e52836731fd9'), 'item': 'mousepad', 'qty': 25, 'size': {'h': 19, 'w': 22.85, 'uom': 'cm'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fda'), 'item': 'notebook', 'qty': 50, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'P'} 

{'_id': ObjectId('63762e071521e52836731fdb'), 'item': 'paper', 'qty': 400, 'size': {'h': 8.5, 'w': 11, 'uom': 'in'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdc'), 'item': 'planner', 'qty': 400, 'size': {'h': 22.85, 'w': 30, 'uom': 'cm'}, 'status': 'D'} 

{'_id': ObjectId('63762e071521e52836731fdd'), 'i

# Do not need to remember each and everything just go to  [Manual](https://www.mongodb.com/docs/manual/)   